Feat/siwe stable

0xtiki opened the pull request:
siwe without web3modal working locally

ken333135 committed:
add 'My account' to user dropdown

ken333135 committed:
add 'POST' method to api proxy

ken333135 committed:
add create acc on BE on login

ken333135 committed:
(feat): add types

ken333135 committed:
(feat): add siwe

ken333135 committed:
add new modules

ken333135 committed:
(feat): connect wallet and verify flow

ken333135 committed:
(fix): assign discordId to session

ken333135 committed:
(fix): discord account creation on discord login

ken333135 committed:
(fix):  move setUser from homepage to navbar

ken333135 committed:
(fix): setUse

ken333135 committed:
(fix): error not being pass through

ken333135 committed:
(refactor): add button color and remove buttons

ken333135 committed:
(refactor): refactor siwe module to new web3 module

ken333135 committed:
(fix): type import error

ken333135 committed:
(feat): add useStrategy and reorg poll form

ken333135 committed:
(feat): poll creation v1

ken333135 committed:
(feat): set provider

0xtiki committed:
fix siwe

0xtiki committed:
fix vercel error

0xtiki committed:
cleanup

0xtiki committed:
lint

0xtiki committed:
Merge branch 'main' into feat/siwe-stable

0xtiki committed:
clean up

0xtiki committed:
fix last 404

0xtiki committed:
poll form improvements

wpmonty commented on hooks/useServer.ts:
```diff
@@ -18,21 +18,294 @@ const MVP_ALLOWED_CHANNELS = {
   "govbot-testing": "959216625180098590"
 };
 
+const channels_response = {
```
bump these two arrays of example data out to a fixtures.ts  file

0xtiki replied:
just saw it now. Probably can get rid of them entirely

wpmonty replied:
just break out for now, replace in new PR

0xtiki committed:
Improvements to poll form:
- relocate single vote check box
- conditionally disable fields based on strategy selection
- set defaults (strategy, block height and single vote checkbox)
- forward BE error messages to poll form
- fix poll author id

wpmonty commented on pages/index.tsx:
```diff
@@ -365,16 +394,4 @@ const Home: NextPage = waitlistDisabled => {
-  const waitlistDisabled =
-    context.query.waitlist === 'true'
-      ? true
-      : process.env.NEXT_PUBLIC_WAITLIST_DISABLED === 'true'
```
why are we removing waitlist override?

wpmonty commented on pages/_app.tsx:
```diff
@@ -17,7 +17,7 @@ function MyApp({ Component, pageProps: { session, ...pageProps } }: AppProps) {
     <SessionProvider session={session}>
       <ChakraProvider theme={theme}>
-        <NavBar waitlistDisabled={pageProps.waitlistDisabled ?? false} />
+        <NavBar />
```
need to support waitlist

wpmonty commented on package.json:
```diff
@@ -25,14 +25,15 @@
     "moment": "^2.29.2",
     "next": "12.1.0",
-    "next-auth": "^4.10.3",
+    "next-auth": "^4.3.2",
```
reduced next-auth version?

wpmonty commented on components/polls/PollForm.tsx:
```diff
@@ -319,10 +376,104 @@ const PollForm: React.FC<BoxProps> = ({ ...props }) => {
+            />
+          </FormControl>
+
+          <FormControl isInvalid={!!errors.block_height?.message}>
```
Hide this field if "Bankless Dao" strategy is not set (the logic should really be "if not token-based strategy".

wpmonty replied:
so we might need a "strategy type" prop in the future, for now we can hard-code

wpmonty commented on components/polls/PollForm.tsx:
```diff
@@ -8,24 +8,32 @@ import {
+import {userAtom} from "../../atoms";
 
-interface Poll {
+const EMPTY_STRATEGY_NAME = 'Empty strategy (returns voting power of 1)';
```
rename "Standard (1 Vote = 1 Vote)"

wpmonty commented on components/polls/PollForm.tsx:
```diff
@@ -319,10 +376,104 @@ const PollForm: React.FC<BoxProps> = ({ ...props }) => {
+
+           <FormControl>
+            <FormLabel mt='1rem' htmlFor='tokenStrategies'>
+              Token Strategy
```
rename label "Voting Strategy"

wpmonty commented on components/polls/PollForm.tsx:
```diff
@@ -319,10 +376,104 @@ const PollForm: React.FC<BoxProps> = ({ ...props }) => {
+            <FormLabel mt='1rem' htmlFor='tokenStrategies'>
+              Token Strategy
+            </FormLabel>
+            <Controller
```
wherever "Bankless Dao" is rename to "BANK - Token Weighted"

wpmonty commented on components/RouteGuard.tsx:
```diff
@@ -13,16 +13,14 @@ const RouteGuard: React.FC = ({ children }) => {
   useEffect(() => {
     if (status === 'unauthenticated') {
-      router.push(url)
+      router.push({ pathname: '/' })
```
revert these to implement waitlist

0xtiki committed:
fix strategies

0xtiki committed:
add route guard

0xtiki committed:
add NavBar.tsx and _app.tsx

0xtiki committed:
add index.tsx from main

0xtiki committed:
update navbar

0xtiki committed:
channels and roles live

0xtiki committed:
fix channel list on polls list page

0xtiki committed:
remove console.log

0xtiki committed:
fix poll results page

0xtiki committed:
sort polls list by date descending

0xtiki committed:
Merge pull request #31 from governatorXYZ/feat/poll-results

Feat/poll results

0xtiki committed:
Merge pull request #30 from governatorXYZ/feat/channel-list

channels and roles live

0xtiki committed:
Merge pull request #29 from governatorXYZ/fix/waitlist

Fix/waitlist

0xtiki committed:
update vote count

0xtiki committed:
Merge pull request #32 from governatorXYZ/feat/results-vote-count

update vote count
